pkg/rancher-desktop/config/settingsImpl.ts

```diff
@@ -353,69 +386,59 @@ export const updateTable: Record<number, (settings: any) => void> = {
-    delete settings.kubernetes.hostResolver;
-    delete settings.kubernetes.checkForExistingKimBuilder;
-    delete settings.kubernetes.memoryInGB;
-    delete settings.kubernetes.numberCPUs;
-    delete settings.kubernetes.suppressSudo;
-    delete settings.kubernetes.WSLIntegrations;
-
-    delete settings.debug;
-    delete settings.pathManagementStrategy;
-    delete settings.telemetry;
-    delete settings.updater;
+    if (_.hasIn(settings, 'kubernetes.suppressSudo')) {
+      _.set(settings, 'application.adminAccess', !settings.kubernetes.suppressSudo);
+      delete settings.kubernetes.suppressSudo;
+    }
+    const replacements: ReplacementDirective[] = [
+      { oldPath: 'debug', newPath: 'application.debug' },
+      { oldPath: 'pathManagementStrategy', newPath: 'application.pathManagementStrategy' },
+      { oldPath: 'telemetry', newPath: 'application.telemetry.enabled' },
+      { oldPath: 'updater', newPath: 'application.updater.enabled' },
+      { oldPath: 'kubernetes.containerEngine', newPath: 'containerEngine.name' },
+      { oldPath: 'kubernetes.experimental.socketVMNet', newPath: 'experimental.virtualMachine.socketVMNet' },
+      { oldPath: 'kubernetes.hostResolver', newPath: 'virtualMachine.hostResolver' },
+      { oldPath: 'kubernetes.memoryInGB', newPath: 'virtualMachine.memoryInGB' },
+      { oldPath: 'kubernetes.numberCPUs', newPath: 'virtualMachine.numberCPUs' },
+      { oldPath: 'kubernetes.WSLIntegrations', newPath: 'WSL.integrations' },
+    ];
+
+    processReplacements(settings, replacements);
+    _.unset(settings, 'kubernetes.checkForExistingKimBuilder');
+    _.unset(settings, 'kubernetes.experimental');
   },
   5: (settings) => {
-    if (settings.containerEngine.imageAllowList) {
-      settings.containerEngine.allowedImages = settings.containerEngine.imageAllowList;
-      delete settings.containerEngine.imageAllowList;
-    }
-    if (settings.virtualMachine.experimental) {
-      if ('socketVMNet' in settings.virtualMachine.experimental) {
-        settings.experimental = { virtualMachine: { socketVMNet: settings.virtualMachine.experimental.socketVMNet } };
-        delete settings.virtualMachine.experimental.socketVMNet;
-      }
+    const replacements: ReplacementDirective[] = [
+      { oldPath: 'autoStart', newPath: 'application.autoStart' },
+      { oldPath: 'hideNotificationIcon', newPath: 'application.hideNotificationIcon' },
+      { oldPath: 'startInBackground', newPath: 'application.startInBackground' },
+      { oldPath: 'window', newPath: 'application.window' },
+      { oldPath: 'containerEngine.imageAllowList', newPath: 'containerEngine.allowedImages' },
+      { oldPath: 'virtualMachine.experimental.socketVMNet', newPath: 'experimental.virtualMachine.socketVMNet' },
+    ];
+
+    processReplacements(settings, replacements);
+    if (settings.virtualMachine?.experimental &&
+      Object.keys(settings.virtualMachine.experimental).length === 0) {
```

Comment:
Since we're using lodash anyway…
```suggestion
    if (_.isEmpty(_.get(settings, 'virtualMachine.experimental'))) {
      _.unset(settings, 'virtualMachine.experimental');
    }
```

Reply:
This makes sense.